py3status/modules/mega.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+"""
+Display status of MEGAcmd.
+
+Configuration parameters:
+    cache_timeout: refresh interval for this module (default 10)
+    format: display format for this module (default "MEGA: {format_sync}")
+    format_sync: display format for every sync (default "{SyncState}")
+    format_sync_separator: show separator if more than one sync (default " ")
+
+Format placeholders:
+    {format_sync} Format for every sync returned by 'mega-sync' command.
+
+format_sync placeholders:
+    Any column returned by 'mega-sync' command - in lower case!
+    For example: id, syncstate, localpath
+
+Requires:
+    MEGAcmd: command-line interface for MEGA
+
+@author Maxim Baz (https://github.com/maximbaz)
+@license BSD
+
+SAMPLE OUTPUT
+{'full_text': 'MEGA: Synced'}
+"""
+
+STRING_NOT_INSTALLED = "MEGAcmd is not installed"
+STRING_NOT_CONFIGURED = "no sync is configured yet"
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    cache_timeout = 10
+    format = "MEGA: {format_sync}"
+    format_sync = "{SyncState}"
+    format_sync_separator = " "
+
+    def post_config_hook(self):
+        if not self.py3.check_commands("mega-sync"):
+            raise Exception(STRING_NOT_INSTALLED)
+
+    def mega(self):
+        output = self.py3.command_output("mega-sync").splitlines()
+        if len(output) == 0:
+            self.py3.error(STRING_NOT_CONFIGURED)
```

Comment:
I think it's wrong to report this  if `mega-sync` is not throwing an error. At this point, `mega-sync` know users are logged in with nothing configured... so it's not really an error. It's your custom error.  We can fix this module a little differently to be more gentle with users and not complaining about this part.

* Users are not signed in. Errors are thrown.
* Users are signed in, but had nothing configured. Be quiet. Show `No MEGA`.
* Users are signed in, with something configured. Show `MEGA {format_output}`

Reply:
See `timewarrior`. Example `format = '[MEGA {format_time}]|No MEGA'`

Reply:
Hmm, but why a string "No MEGA" in successful color is better than a string "you don't have sync, configure one" in an error color? The latter one draws your attention and specifically describes what you need to do to get the module working, "No MEGA" would have been a very confusing output for me.

Reply:
What successful color? Default color? You view this as something important. I view this as a simple number of syncs, file paths, or updates. I think it is fine to have none.

We will throw error if users don't have this installed. After that, `mega-sync` will throw errors if users are not logged in... or when things does not go its way. That's when we can deter responsibility to `mega-sync` and not having to make custom errors... And also because we don't know how `mega-sync` work.

To reiterate...
* Not installed? We throw an error.
* Users not signed in? `mega-sync` throw an error. We deter everything here.
* Users signed in + no number of syncs... `mega-sync` does not throw an error.
* Users signed in, but no output? `mega-sync` does not throw an error, but you want to...
* Users does not have Internet, `mega-sync` does not throw an error, but your code might...

We also don't throw errors on other things such as...

* No Internet connection on `whatismyip`.
* Audio players not configured or running (iirc mpd throws error, grrr.)
* No new mails, timew intervals, tasks, updates, etc.
* et cetera

Generally, we want to go for fewer configs, fewer errors, fewer issues, etc.

Users seeing `No MEGA` can figure out their issues fast because they're smart cookies and they already run linux, i3wm, and py3status too... Also because `mega-sync` does not consider this a fatal error in first place so basically, both `mega-sync` and we are saying that users can set this up later when they want to... instead of trying to make an error on possibly something trivial.

Reply:
OK, makes sense.